core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -669,6 +753,27 @@ private[spark] class ExecutorAllocationManager(
     private val stageAttemptToExecutorPlacementHints =
       new mutable.HashMap[StageAttempt, (Int, Map[String, Int], Int)]
 
+    // to track total no. of tasks in each stage of a micro-batch (streaming use case)
+    // this will help in requesting resources by counting pending tasks in job,
+    // rather than counting pending tasks in a stage.
+    private val stageIdToNumTasks = new mutable.HashMap[Int, Int]
```

Comment:
Are there any race conditions for updating this map?  I am not an expert on the DAGScheduler or ExecutorAllocationManager, but it seems to be more then one thread will be reading/updating this map.

Reply:
like much of the data structures in ExecutorAllocationListener this  map will only be updated by spark events. In this case `onJobStart`, `onJobEnd` and `onStageSubmitted`

Reply:
I have updated all the touchpoints with `allocationManager.synchronized`. Thanq